Remove BoxerComponent

MWKane opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
*This is my first PR.*

Resolves #13848.
- Removes BoxingComponent thus removing all combat bonuses from Boxers

<!-- What does it change? What other things could this impact? -->


**Media**
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged.

Only put changes that are visible and important to the player on the changelog.

Don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers

Putting a name after the :cl: symbol will change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB
-->

:cl: TimmyTwoHands
- tweak: Removed combat bonuses from Boxers.

GoodWheatley commented:
The stamina damage modifier should probably be removed, it’s going to mess with balancing boxing gloves if the stamina values on the glove prototype are multiplied by 1.75 in the game

MWKane commented:
> The stamina damage modifier should probably be removed, it’s going to mess with balancing boxing gloves if the stamina values on the glove prototype are multiplied by 1.75 in the game

I've modified the combat values in gloves.yml and removed the component range/stamina modifiers. Since that was all BoxingGlovesComponent/BoxingSystem were doing, I've removed them completely.

EmoGarbage404 commented:
boxing glove component was meant to be on specific characters to give them a combat bonus. Adding in the bonus on the weapon and adjusting it's damage values as well is pointless. Just remove the damage bonus and keep the weapon stats as-is.

EmoGarbage404 commented:
nvm, you already did that.

MWKane commented:
> boxing glove component was meant to be on specific characters to give them a combat bonus. Adding in the bonus on the weapon and adjusting it's damage values as well is pointless. Just remove the damage bonus and keep the weapon stats as-is.

I guess I should've asked for clarification before starting on this. I'm a little confused now. I'm having trouble reconciling your comment and the original remarks in #13848 
> Boxers should have the same combat stats as everyone else, just leave it on the gloves.

Do we want the component on the job or the gloves?

EmoGarbage404 commented:
> Do we want the component on the job or the gloves?

just remove the component and leave the gloves the same.

MWKane commented:
To sum it up, I have:

- Removed BoxerComponent
- Removed any component bonuses from boxing gloves, removing the need for BoxingGlovesComponent and BoxingSystem
- Left the combat stats on boxing gloves the same (giving them an effective nerf of -0.75 range and -6 stamina per hit)

Is this what you wanted @EmoGarbage404?

EmoGarbage404 reviewed (approved):
Looks good to me!

moonheart08 commented:
why remove the boxer bonus though?

GoodWheatley commented:
> why remove the boxer bonus though?

Because that’s what Sloth suggested in the linked issue, that the boxer bonus be moved to the gloves themselves. The author of the PR is getting confused because we’re giving him conflicting instructions

AJCM-git commented:
is the description correct? i see no BoxingGlovesComponent changes in the code

MWKane commented:
@AJCM-git The contents of the PR changed based on the feedback I received - I've updated the title/description to reflect the changes.
